internal/xds/v3/snapshotter.go

```diff
@@ -37,16 +38,19 @@ type snapshotter struct {
 
 func (s *snapshotter) Generate(version string, resources map[envoy_types.ResponseType][]envoy_types.Resource) error {
 	// Create a snapshot with all xDS resources.
-	snapshot := envoy_cache_v3.NewSnapshot(
+	snapshot, err := envoy_cache_v3.NewSnapshot(
 		version,
-		resources[envoy_types.Endpoint],
-		resources[envoy_types.Cluster],
-		resources[envoy_types.Route],
-		resources[envoy_types.Listener],
-		nil,
-		resources[envoy_types.Secret],
-		nil,
+		map[envoy_resource_v3.Type][]envoy_types.Resource{
+			envoy_resource_v3.EndpointType: resources[envoy_types.Endpoint],
+			envoy_resource_v3.ClusterType:  resources[envoy_types.Cluster],
+			envoy_resource_v3.RouteType:    resources[envoy_types.Route],
+			envoy_resource_v3.ListenerType: resources[envoy_types.Listener],
+			envoy_resource_v3.SecretType:   resources[envoy_types.Secret],
+		},
```

Comment:
Looks like we could probably update this method signature to take in a `map[envoy_resource_v3.Type][]envoy_types.Resource`, and update the caller to pass that in. Then could use `resources` directly here.

Reply:
ah yeah let me rework that, theres some room for improvement and de-duplication